streamlit_app/components/subcomponents/matplotlib_presets.py

```diff
@@ -0,0 +1,245 @@
+import math
+
+WIN_RATE = """
+plt.subplot({nrows}, {ncols}, {next_plot_idx})
+df[df.adv_color == "b"].groupby("adv_steps").mean().adv_win.plot(
+    label="adv = black"
+)
+df[df.adv_color == "w"].groupby("adv_steps").mean().adv_win.plot(
+    label="adv = white"
+)
+plt.ylabel("Win rate")
+plt.title("win rate")
+plt.legend()
+"""
+
+GAME_COUNT = """
+plt.subplot({nrows}, {ncols}, {next_plot_idx})
+df[df.adv_color == "b"].groupby("adv_steps").count().adv_win.plot(
+    label="adv = black"
+)
+df[df.adv_color == "w"].groupby("adv_steps").count().adv_win.plot(
+    label="adv = white"
+)
+plt.ylabel("# of games")
+plt.title("game count")
+plt.legend()
+"""
+
+SCORE_EVOLUTION_MEDIAN = """
+plt.subplot({nrows}, {ncols}, {next_plot_idx})
+df[df.adv_color == "b"].groupby("adv_steps").median().adv_minus_victim_score.plot(
+    label="adv = black"
+)
+df[df.adv_color == "w"].groupby("adv_steps").median().adv_minus_victim_score.plot(
+    label="adv = white"
+)
+plt.ylabel("adv_minus_victim_score")
+plt.title("score evolution (median)")
+plt.legend()
+"""
+
+NUM_MOVES_EVOLUTION_MEDIAN = """
+plt.subplot({nrows}, {ncols}, {next_plot_idx})
+df[df.adv_color == "b"].groupby("adv_steps").median().num_moves.plot(
+    label="adv = black"
+)
+df[df.adv_color == "w"].groupby("adv_steps").median().num_moves.plot(
+    label="adv = white"
+)
+plt.ylabel("num_moves")
+plt.title("# of moves evolution (median)")
+plt.legend()
+"""
+
+SCORE_EVOLUTION = """
+ax = plt.subplot({nrows}, {ncols}, {next_plot_idx}, projection="3d")
+hist_3d(
+    df,
+    ts=np.sort(df.adv_steps.unique())[:-2],
```

Comment:
Good point, thanks. This code is copied from Tony's notebooks. I think I'm going to rewrite the way that the source for these figures is stored, so that these types of decisions don't have spill-over effects.